Source/Section/PDFSectionObject.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  PDFSectionObject.swift
+//  TPPDF
+//
+//  Created by Marco Betschart on 05.05.18.
+//
+
+class PDFSectionObject: PDFObject {
+
+	var section: PDFSection
+
+	init(section: PDFSection) {
+		self.section = section
+	}
+
+	override func calculate(generator: PDFGenerator, container: PDFContainer) throws -> [(PDFContainer, PDFObject)] {
+		var result: [(PDFContainer, PDFObject)] = []
+
+		let originalLeftIndent = [
+			PDFContainer.contentLeft: generator.layout.indentation.leftIn(container: .contentLeft),
+			PDFContainer.contentCenter: generator.layout.indentation.leftIn(container: .contentCenter),
+			PDFContainer.contentRight: generator.layout.indentation.leftIn(container: .contentRight)
+		]
+
+		let originalRightIndent = [
+			PDFContainer.contentLeft: generator.layout.indentation.rightIn(container: .contentLeft),
+			PDFContainer.contentCenter: generator.layout.indentation.rightIn(container: .contentCenter),
+			PDFContainer.contentRight: generator.layout.indentation.rightIn(container: .contentRight)
+		]
+
+		let originalContentOffset = generator.getContentOffset(in: container)
+		let contentWidth = generator.document.layout.width - generator.document.layout.margin.left - generator.document.layout.margin.right
```

Comment:
Try using `generator.document.layout.contentSize.width` instead of calculate content width your self

Reply:
awesome, thanks for the hint! Fixed.